hazelcast/src/main/java/com/hazelcast/config/JoinConfig.java

```diff
@@ -218,35 +220,47 @@ public boolean isAutoDetectionEnabled() {
      */
     @SuppressWarnings("checkstyle:npathcomplexity")
     public void verify() {
+        List<String> enabledDiscoveries = new ArrayList<>();
         int countEnabled = 0;
         if (getTcpIpConfig().isEnabled()) {
             countEnabled++;
+            enabledDiscoveries.add("TCP/IP");
         }
         if (getMulticastConfig().isEnabled()) {
             countEnabled++;
+            enabledDiscoveries.add("Multicast");
         }
         if (getAwsConfig().isEnabled()) {
             countEnabled++;
+            enabledDiscoveries.add("AWS discovery");
         }
         if (getGcpConfig().isEnabled()) {
             countEnabled++;
+            enabledDiscoveries.add("GCP discovery");
         }
         if (getAzureConfig().isEnabled()) {
             countEnabled++;
+            enabledDiscoveries.add("Azure discovery");
         }
         if (getKubernetesConfig().isEnabled()) {
             countEnabled++;
+            enabledDiscoveries.add("Kubernetes discovery");
         }
         if (getEurekaConfig().isEnabled()) {
             countEnabled++;
+            enabledDiscoveries.add("Eureka discovery");
         }
 
         Collection<DiscoveryStrategyConfig> discoveryStrategyConfigs = discoveryConfig.getDiscoveryStrategyConfigs();
-        countEnabled += discoveryStrategyConfigs.size();
+        if (!discoveryStrategyConfigs.isEmpty()) {
+            countEnabled += discoveryStrategyConfigs.size();
```

Comment:
One more question. Shouldn't this be counted only as one even if the `size() > 1`?

Reply:
Yes, I believe it should be counted as only 1.

Reply:
Fixed.